src/sentry/migrations/0748_fix_substatus_for_ignored_groups.py

```diff
@@ -0,0 +1,107 @@
+# Generated by Django 5.0.7 on 2024-08-05 17:50
+
+from django.apps.registry import Apps
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+# Copying constants defined in the models
+
+
+class ActivityType:
+    SET_IGNORED = 3
+
+
+class GroupHistoryStatus:
+    ARCHIVED_UNTIL_ESCALATING = 15
+    ARCHIVED_FOREVER = 16
+    ARCHIVED_UNTIL_CONDITION_MET = 17
+
+
+class GroupSubStatus:
+    UNTIL_ESCALATING = 1
+    # Group is ignored/archived for a count/user count/duration
+    UNTIL_CONDITION_MET = 4
+    # Group is ignored/archived forever
+    FOREVER = 5
+
+
+class GroupStatus:
+    IGNORED = 2
+
+
+# End copy
+
+ACTIVITY_DATA_FIELDS = {
+    "ignoreCount",
+    "ignoreDuration",
+    "ignoreUntil",
+    "ignoreUserCount",
+    "ignoreUserWindow",
+    "ignoreWindow",
+}
+
+
+def backfill_substatus_for_ignored_groups(
+    apps: Apps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Group = apps.get_model("sentry", "Group")
+    Activity = apps.get_model("sentry", "Activity")
+    GroupSnooze = apps.get_model("sentry", "GroupSnooze")
+
+    groups = Group.objects.filter(status=GroupStatus.IGNORED, substatus=None)
+    activity = Activity.objects.filter(group__in=groups, type=ActivityType.SET_IGNORED)
+
+    for group in groups:
+        group_activity = activity.filter(group_id=group.id).order_by("-datetime").first()
+        new_substatus = None
+        if group_activity:
+            # If ignoreUntilEscalating is set, we should set the substatus to UNTIL_ESCALATING
+            if group_activity.data.get("ignoreUntilEscalating", False):
+                new_substatus = GroupSubStatus.UNTIL_ESCALATING
+            # If any other field in the activity data is set, we should set the substatus to UNTIL_CONDITION_MET
+            elif any(group_activity.data.get(field) for field in ACTIVITY_DATA_FIELDS):
+                new_substatus = GroupSubStatus.UNTIL_CONDITION_MET
+
+        # If no activity is found or the activity data is not set, check the group snooze table
+        if not new_substatus:
+            snooze = GroupSnooze.objects.filter(group=group)
+            if snooze.exists():
+                # If snooze exists, we should set the substatus to UNTIL_CONDITION_MET
+                new_substatus = GroupSubStatus.UNTIL_CONDITION_MET
+            else:
+                # If we have no other information stored about the group's status conditions, the group is ignored forever
+                new_substatus = GroupSubStatus.FOREVER
+
+        group.substatus = new_substatus
+
+    Group.objects.bulk_update(groups, ["substatus"])
```

Comment:
How many groups are we going to have here? We'll end up ooming if it's too many

Reply:
We already have an index on `(status, substatus, id)` so that'll actually be fine with the `RangeQuerySetWrapperWithProgressBarApprox`, so I think we should use that. If you want to make it simpler, you can just process a row at a time and skip batching - it'll take less than an hour for 123k rows.